contracts/contract/token/RocketTokenRETH.sol

```diff
@@ -135,4 +136,21 @@ contract RocketTokenRETH is RocketBase, ERC20, RocketTokenRETHInterface {
         rocketDepositPool.withdrawExcessBalance(_ethRequired.sub(ethBalance));
     }
 
+    // This is called by the base ERC20 contract before all transfer, mint, and burns
+    function _beforeTokenTransfer(address from, address, uint256) internal override {
+        // Don't run check if this is a mint transaction
+        if (from != address(0)) {
+            // Check which block the user's last deposit was
+            bytes32 key = keccak256(abi.encodePacked("user.deposit.block", from));
+            uint256 lastDepositBlock = getUint(key);
+            if (lastDepositBlock > 0) {
+                // Ensure enough blocks have passed
+                RocketDAOProtocolSettingsNetworkInterface rocketDAOProtocolSettingsNetwork = RocketDAOProtocolSettingsNetworkInterface(getContractAddress("rocketDAOProtocolSettingsNetwork"));
+                uint256 blocksPassed = block.number.sub(lastDepositBlock);
+                require(blocksPassed > rocketDAOProtocolSettingsNetwork.getRethDepositDelay(), "Not enough time has passed since deposit");
```

Comment:
Should we check if `getRethDepositDelay()` is 0 before running any of the logic in this function? If we disable it by setting it to 0 currently, it'll still execute all this logic before passing. Might save some gas to just check if it's 0 first?